src/include/homestore/btree/detail/btree_node.hpp

```diff
@@ -480,13 +510,11 @@ class BtreeNode : public sisl::ObjLifeCounter< BtreeNode > {
         return (edge_id() != empty_bnodeid);
     }
 
+    void set_modified_cp_id(int64_t id) { get_persistent_header()->modified_cp_id = id; }
     friend void intrusive_ptr_add_ref(BtreeNode* node) { node->m_refcount.increment(1); }
 
     friend void intrusive_ptr_release(BtreeNode* node) {
-        if (node->m_refcount.decrement_testz(1)) {
-            node->~BtreeNode();
-            delete[] uintptr_cast(node);
-        }
+        if (node->m_refcount.decrement_testz(1)) { delete node; }
```

Comment:
We no longer doing placement new for btree node, so no need of explicit destructor.